src/jsonLogic.js

```diff
@@ -203,3 +245,119 @@ function handleNestedObjectForComputedValues(values, formValues, parentID, logic
     })
   );
 }
+
+/**
+ * Builds a sample empty object for the given schema.
+ *
+ * Recursively builds an object with empty values for each property in the schema.
+ * Used to provide a valid data structure to test jsonLogic validation rules against.
+ *
+ * Handles objects, arrays, and nested schemas.
+ *
+ * @param {Object} schema - The JSON schema
+ * @returns {Object} Sample empty object based on the schema
+ */
+function buildSampleEmptyObject(schema = {}) {
+  const sample = {};
+  if (typeof schema !== 'object' || !schema.properties) {
+    return schema;
+  }
+
+  for (const key in schema.properties) {
+    if (schema.properties[key].type === 'object') {
+      sample[key] = buildSampleEmptyObject(schema.properties[key]);
+    } else if (schema.properties[key].type === 'array') {
+      const itemSchema = schema.properties[key].items;
+      sample[key] = buildSampleEmptyObject(itemSchema);
+    } else {
+      sample[key] = true;
+    }
+  }
+
+  return sample;
+}
+
+/**
+ * Validates inline jsonLogic rules defined in the schema's x-jsf-logic-computedAttrs.
+ *
+ * For each field with computed attributes, checks that the variables
+ * referenced in the rules exist in the schema.
+ *
+ * Throws if any variable in a computed attribute rule does not exist.
+ *
+ * @param {Object} jsonSchema - The JSON schema object
+ * @param {Object} sampleEmptyObject - Sample empty object based on the schema
+ */
+function validateInlineRules(jsonSchema, sampleEmptyObject) {
+  const properties = (jsonSchema?.properties || jsonSchema?.items?.properties) ?? {};
+  Object.entries(properties)
```

Comment:
Nit: I think this may be easier to read if it were done recursively?

Reply:
this is what I tried but it wasn't working as I was expecting:

```

function validateInlineRules(jsonSchema, sampleData, parentKey = '') {
  const properties = (jsonSchema?.properties || jsonSchema?.items?.properties) ?? {};
  
  Object.keys(properties).forEach((key) => {
    const fieldKey = parentKey ? `${parentKey}.${key}` : key;

    if (properties[key]['x-jsf-logic-computedAttrs']) {
      // validate computedAttrs rules for this field

      Object.values(properties[key]['x-jsf-logic-computedAttrs']).forEach((rule) => {
        checkRuleIntegrity(
          rule,
          fieldKey,
          sampleData,
          (item) =>
            `[json-schema-form] json-logic error: fieldName "${item.var}" doesn't exist in field "${fieldKey}.x-jsf-logic-computedAttrs.${key}".`
        );
      });
    }

    if (properties[key].type === 'object') {
      // recurse to validate nested object fields
      validateInlineRules(properties[key], sampleData, fieldKey);
    }

    if (properties[key].type === 'array') {
      // recurse to validate array item fields
      validateInlineRules(properties[key].items, sampleData, fieldKey);
    }
  });
}
```

I'm not gonna consider this a blocker since its a nit, but if you see what I'm doing wrong, lmk 😂